test/test.js

```diff
@@ -0,0 +1,128 @@
+var server = require('../app/server');
+var assert = require('assert');
+var http = require('http');
+var request = require('supertest');
+var expect = require('expect.js');
+var url = 'localhost:9000';
+
+//npm install mocha, superagent, supertest, expect.js
+
+describe('server', function () {
+    before(function (done) {
+        done();
+    });
+
+    // after(function (done) {
+    //     server.close();
+    //     done();
+    // });
+});
+
+describe('/api/attribute', function () {
+    var id;
+
+   //getting all attributes
+    it('should reply with HTTP status code 200 when the correct email and ' + 
+        'user headers are set', function(done) {
+        request(url)
+            .get('/api/attribute')
+            .set('x-forwarded-email', 'anton.lundin2@softhouse.se')
+            .set('x-forwarded-user', 'Anton Lundin2')
+            .set('Content-Type', 'application/json')
+            .send()
+            // end handles the response
+            .end(function(err, res) {
+                if (err) {
+                    throw err;
+                }
+                expect(res).to.exist;
+                expect(res.status).to.equal(200);
+                done();
+            });
+    });
+
+    //posting an attribute
+    it('should respond with a correctly formatted json object containing the posted attribute "test1"', function(done) {
+        request(url)
+            .post('/api/attribute')
+            .set('x-forwarded-email', 'anton.lundin2@softhouse.se')
+            .set('x-forwarded-user', 'Anton Lundin2')
+            .set('Content-Type', 'application/json')
+            .send({
+                name: 'test1'
+            })
+            // end handles the response
+            .end(function(err, res) {
+                if (err) {
+                    throw err;
+                }
+                expect(res).to.exist;
+                expect(res.status).to.equal(200);
+                expect(Object.keys(res.body).length).to.equal(4);
+
+                expect(res.body).to.have.key('createdAt');
+                expect(res.body).to.have.key('updatedAt');
+                expect(res.body).to.have.key('_id');
+                expect(res.body).to.have.key('name');
+
+                expect(res.body.name).to.equal('test1');
+                expect("0x" + res.body._id).to.be.within(0, Infinity);
+
+                id = res.body._id;
+                done();
+            });
+    });
+
+    //retrieving previously posted attribute by its id
+    it('should respond with a correctly formatted json object containing the previously posted attribute "test1"', function(done) {
```

Comment:
I don't think there are any guarantees about what order the tests are run in. And even if there is, I don't want the testcases to depend on each other. If a single test fails I want to only have to rerun that one testcase when working on fixing it.